### PR TITLE
Mox Lesson + Testing Lesson Updates

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
This PR is opened in favor of this one from @fireproofsocks. It:

* Moves @fireproofsocks' new content on Mox to its own lesson under `libraries/`
* Update existing "mocking" section of Testing lesson to clarify and link to Mox lesson

### CODE REVIEW

File: en/lessons/libraries/mox.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+defmodule MyAppTest do
+  use ExUnit.Case
+  # 1. Import Mox
+  import Mox
+  # 2. setup fixtures
+  setup :verify_on_exit!
+
+  test ":ok on 200" do
+    HTTPoison.BaseMock
+    |> expect(:get, fn _ -> {:ok, "What a guy!"} end)
```
Comment (doomspork) [inline en/lessons/libraries/mox.md:179]: We don't use pipelines with only a single function call in our other examples so let's stick with that :+1: 

```suggestion
    expect(HTTPoison.BaseMock, :get, fn _ -> {:ok, "What a guy!"} end)
```

File: en/lessons/libraries/mox.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    assert {:ok, _} = MyModule.get_username("twinkie")
+  end
+
+  test ":error on 404" do
+    HTTPoison.BaseMock
+    |> expect(:get, fn _ -> {:error, "Sorry!"} end)
```
Comment (doomspork) [inline en/lessons/libraries/mox.md:186]:
```suggestion
    expect(HTTPoison.BaseMock, :get, fn _ -> {:error, "Sorry!"} end)
```

**Comment by doomspork** [comment]
@fireproofsocks do you have a twitter handle? I'd love to give you a shoutout when we announce this new lesson you've put together.

**Comment by fireproofsocks** [comment]
Thanks @doomspork ... although I have the `@fireproofsocks` handle on Twitter, I rarely use it. I also have https://medium.com/@fireproofsocks but most of the articles I write are paid contributions elsewhere, so there's not much to look at there, but I guess it wouldn't hurt to reference them.  Thanks!

**Pull request merged**